pkg/operator/controller/gatewayapi/rbac.go

```diff
@@ -0,0 +1,123 @@
+package gatewayapi
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+var managedClusterRoles = []*rbacv1.ClusterRole{
+	manifests.GatewayAPIAdminClusterRole(),
+	manifests.GatewayAPIViewClusterRole(),
+}
+
+// ensureGatewayAPIRBAC ensures that all RBAC resources for Gateway API are
+// deployed to the cluster, and up to date.
+func (r *reconciler) ensureGatewayAPIRBAC(ctx context.Context) error {
+	var errs []error
+
+	for _, clusterRole := range managedClusterRoles {
+		if err := r.ensureClusterRole(ctx, clusterRole); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+// ensureClusterRole ensures that a specific ClusterRole RBAC resource is
+// deployed to the cluster, and up to date.
+func (r *reconciler) ensureClusterRole(ctx context.Context, desired *rbacv1.ClusterRole) error {
+	clusterRoleAlreadyExists, current, err := r.currentClusterRole(ctx, desired.Name)
+	if err != nil {
+		return err
+	}
+
+	if clusterRoleAlreadyExists {
+		if updated, err := r.updateClusterRole(ctx, current, desired); err != nil {
+			return err
+		} else if updated {
+			if _, _, err := r.currentClusterRole(ctx, desired.Name); err != nil {
+				return err
+			}
+		}
+	} else {
+		return r.createClusterRole(ctx, desired)
+	}
+
+	return nil
+}
+
+// currentClusterRole retrieves a ClusterRole from the cluster, given the name
+// of that role.
+func (r *reconciler) currentClusterRole(ctx context.Context, name string) (bool, *rbacv1.ClusterRole, error) {
+	var existingClusterRole rbacv1.ClusterRole
+
+	if err := r.client.Get(ctx, types.NamespacedName{Name: name}, &existingClusterRole); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, fmt.Errorf("failed to get ClusterRole %s: %w", name, err)
+	}
+
+	return true, &existingClusterRole, nil
+}
+
+// createClusterRole creates the desired ClusterRole on the cluster.
+func (r *reconciler) createClusterRole(ctx context.Context, desired *rbacv1.ClusterRole) error {
+	desired.ResourceVersion = ""
```

Comment:
Why is this needed?  The object came from the manifest, so it won't have `revisionVersion` set anyway, right?